node_exporter.go

```diff
@@ -97,20 +99,6 @@ func main() {
 		log.Infof(" - %s", n)
 	}
 
-	http.HandleFunc(*metricsPath, handler)
-	http.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
-		w.Write([]byte(`<html>
-			<head><title>Node Exporter</title></head>
-			<body>
-			<h1>Node Exporter</h1>
-			<p><a href="` + *metricsPath + `">Metrics</a></p>
-			</body>
-			</html>`))
-	})
-
-	log.Infoln("Listening on", *listenAddress)
-	err = http.ListenAndServe(*listenAddress, nil)
-	if err != nil {
-		log.Fatal(err)
-	}
+	// Use our shared code to run server and exit on error. Upstream's code below will not be executed.
+	exporter_shared.RunServer("Node", *listenAddress, *metricsPath, http.HandlerFunc(handler))
```

Comment:
Please move that line above, and do not delete that block of code. Also, please read the second sentence of the comment you copied. 😄

Yes, it will be dead code. Yes, I know that sounds stupid. 😄 The idea is to have a minimal diff/patch to simplify future updates. If upstream changes lines above, we will have a merge conflict. If we do not remove then, a merge can be performed automatically and without much thinking/checking.

Reply:
> ...and do not delete that block of code
> Yes, it will be dead code. Yes, I know that sounds stupid.
It was I, sorry. ¯\_(ツ)_/¯

Reply:
It has been placed back.